src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -56,6 +58,9 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             {
                 return await ProcessRequestAsync(req, function, ct);
             };
+
+            IFuncExecutor proxyFunctionExecutor = new ProxyFunctionExecutor(this._scriptHostManager, controllerContext.Configuration.DependencyResolver, this);
```

Comment:
>this [](start = 153, length = 4)

I think passing this controller instance in here is incorrect. 
Say request #1 is the initial call to the proxy. That has a backend call to a custom function (and if I recall, may invoke multiple custom functions for a single request) 
So that could lead to subsequent request #2, #3 for Functions. 

Controllers are specific to a request.  This controller instance is for the incoming request (#1). But you're reusing it for request #2, #3 too. 
 
I think the correct think here is to split this controller out. Create a new non-controller class (ie, "FunctionRequestInvoker")  which has ProcessRequestAsync() below. 
Then both this controller and your IFuncExecutor can just calls that.

Reply:
I was trying to minimize changes to the FunctionController. I can make a class FunctionRequestInvoker and move some common logic there as static.

---
In reply to: [136211006](https://github.com/Azure/azure-webjobs-sdk-script/pull/1839#discussion_r136211006) [](ancestors = 136211006)